File: javascript/maxProfit.js
'use strict';

/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    let result = 0;
    for (let i = 0; i < prices.length; i++) {
        if (prices[i] < prices[i + 1]) {
            result += prices[i + 1] - prices[i];
        }
    }
    return result < 0 ? 0 : result;
};

File: golang/maxProfit.go
package solution

func maxProfit(prices []int) int {
	result := 0
	length := len(prices) - 1
	for i, v := range prices {
		if i != length && v < prices[i + 1] {
			result += prices[i + 1] - v
		}
	}
	return result
}